### PR TITLE
Filter unsupported endpoints for delete command

### PR DESCRIPTION
**Pull request by jskelin**
#### What this PR does / Why we need it:
When the new endpoint is under development, it is hidden behind the feature flag `MONACO_EXPERIMENTAL`. With this change, endpoints under development will also be unsupported for the `delete` command unless `MONACO_EXPERIMENTAL` ff is present.

#### Special notes for your reviewer:
The main change is in `pkg/delete/delete_loader.go`. Since `delete` is used in almost every integration test for the cleanup process, necessary changes are done in the rest of the code.

### CODE REVIEW

File: cmd/monaco/integrationtest/v2/all_configs_integration_test.go [hunk 1]
```diff
@@ -19,52 +19,49 @@
...
+		t.Run(tc.name, func(t *testing.T) {
+			configFolder := "test-resources/integration-all-configs/"
+			manifest := configFolder + "manifest.yaml"
+
+			envVars := map[string]string{featureflags.Experimental().EnvName(): "true"}
+
+			RunIntegrationWithCleanupGivenEnvs(t, configFolder, manifest, tc.environment, "AllConfigs", envVars, func(fs afero.Fs, _ TestContext) {
+				{
+					// This causes a POST for all configs:
+					_, err := test.Monacof("deploy %s --environment %s", manifest, tc.environment).WithFs(fs).Run()
```
Comment (UnseenWizzard) [inline cmd/monaco/integrationtest/v2/all_configs_integration_test.go:54]: nit/thought: this change isn't really related to this commit, I think they should have gone in the refactor commit - having these additional changes makes it a bit harder to review the actual change.

File: pkg/delete/delete_loader.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func LoadEntriesToDelete(fs afero.Fs, deleteFile string) (DeleteEntries, error)
 	context := &loaderContext{
 		fs:         fs,
 		deleteFile: filepath.Clean(deleteFile),
-		knownApis:  api.NewAPIs(),
+		knownApis:  api.NewAPIs().Filter(api.RemoveDisabled),
```
Comment (UnseenWizzard) [inline pkg/delete/delete_loader.go:74]: This single line the 'real' fix, right? I'm okay adding refactoring changes to a PR, but as you already started, it would make it easier for me to review if things are nicely split into separate commits for refactoring and the change.

File: pkg/delete/delete_loader_test.go [hunk 3]
```diff
@@ -173,6 +173,7 @@ func TestParseDeleteFileDefinitionsWithInvalidDefinition(t *testing.T) {
 }
 
 func TestLoadEntriesToDelete(t *testing.T) {
+	t.Setenv(featureflags.Experimental().EnvName(), "true")
```
Comment (UnseenWizzard) [inline pkg/delete/delete_loader_test.go:176]: might have missed it, but could you please add covering the fact that delete filters out disabled APIs? So this can't accidentally break in the future. 
E.g. have a test that sets up a dedicated API map with an experimental one (so it's not impacted by setting current ones to non-experimental), then assert that there no delete entries loaded from a file that defines that API.

File: cmd/monaco/internal/test/cmd.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	"regexp"
+	"strings"
+)
+
+type Cmd struct {
+	cmd string
+	fs  afero.Fs
+}
+
+func Monacof(cmd string, args ...any) *Cmd {
```
Comment (Laubi) [inline cmd/monaco/internal/test/cmd.go:32]: Please add documentation to all public symbols where it's not obvious what it does. 
Especially Monacof and Cmd are not clear; WithFs, Run, etc are clear

File: cmd/monaco/internal/test/cmd.go
Comment (Laubi) [inline cmd/monaco/internal/test/cmd.go:1]: This file should still be in /cmd/monaco/integrationtest, as it's only used there, right?

File: cmd/monaco/internal/test/cmd.go (hunk 4 above)
Comment (Laubi) [inline cmd/monaco/internal/test/cmd.go:32]: `RunMonaco` imo would be better (?)

File: cmd/monaco/internal/test/cmd.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	cmd = fmt.Sprintf(cmd, args...)
+
+	cmd = regexp.MustCompile(`\s+`).ReplaceAllString(cmd, " ")
```
Comment (Laubi) [inline cmd/monaco/internal/test/cmd.go:35]: Please add a brief documentation what and **why** you're doing this here.

To the what: I think replacing all spaces with one space is not intended, we should not have that. E.g. passing a file with multiple empty spaces would become invalid.

e.g. `deploy "man  ifest.yaml"` would become `deploy "man ifest.yaml"`, right?

File: cmd/monaco/internal/test/cmd.go [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+func (cmd *Cmd) WithFs(fs afero.Fs) *Cmd {
+	cmd.fs = fs
+	return cmd
+}
+
+func (cmd *Cmd) Run() (afero.Fs, error) {
+	fs := cmd.fs
+	if fs == nil {
+		fs = afero.NewCopyOnWriteFs(afero.NewOsFs(), afero.NewMemMapFs())
```
Comment (Laubi) [inline cmd/monaco/internal/test/cmd.go:50]: We have a test util for that, and please, try to use the TempFs here instead

File: cmd/monaco/internal/test/cmd.go (hunk 4 above)
Comment (warber) [reply cmd/monaco/internal/test/cmd.go:32]: Then it would be `RunMonaco(...).Run()`
